File: guide-to-jakarta-ee-with-react-and-postgresql/src/main/java/de/rieckpil/blog/BookResource.java
package de.rieckpil.blog;

import org.eclipse.microprofile.config.inject.ConfigProperty;

import javax.inject.Inject;
import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import java.util.List;

@Path("books")
public class BookResource {

    @Inject
    @ConfigProperty(name = "book_list_size", defaultValue = "10")
    private Integer bookListSize;

    @PersistenceContext
    private EntityManager entityManager;

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Response getAllBooks() {

        List<Book> allBooks = this.entityManager
                .createQuery("SELECT b FROM Book b", Book.class)
                .setMaxResults(bookListSize)
                .getResultList();

        return Response.ok(allBooks).build();
    }
}

File: java-ee-with-kotlin/src/main/kotlin/de.rieckpil.blog/BookResource.kt
package de.rieckpil.blog

import javax.inject.Inject
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.PathParam
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("books")
@Produces(MediaType.APPLICATION_JSON)
class BookResource {

    @Inject
    private lateinit var bookService: BookService;

    @GET
    fun getAllBooks(): Response = Response.ok(bookService.getAllBooks()).build()

    @GET
    @Path("/{id}")
    fun getBookById(@PathParam("id") id: Long): Response {
        var book: Book? = bookService.getBookById(id) ?: return Response.status(Response.Status.NOT_FOUND).build()
        return Response.ok(book).build()
    }
}